server/protocol/protocol.py

```diff
@@ -4,10 +4,26 @@
 from asyncio import StreamReader, StreamWriter
 
 import server.metrics as metrics
+from server.config import config
 
 from ..asyncio_extensions import synchronizedmethod
 
-json_encoder = json.JSONEncoder(separators=(",", ":"))
+
+class CustomJSONEncoder(json.JSONEncoder):
+    # taken from https://stackoverflow.com/a/53798633
+    def encode(self, o):
+        def round_floats(o):
+            if isinstance(o, float):
+                return round(o, config.JSON_FLOAT_MAX_DIGITS)
+            if isinstance(o, dict):
+                return {k: round_floats(v) for k, v in o.items()}
+            if isinstance(o, (list, tuple)):
+                return [round_floats(x) for x in o]
+            return o
+        return super().encode(round_floats(o))
```

Comment:
Maybe we should add a config option to disable this feature just in case it turns out to be too costly in production. Although, I think even if it doubles the json encoding time that will still be fine judging by the profiling information I've collected from the server in the past.

Reply:
Yes, sure, what would you call it?

Reply:
I think `JSON_ROUND_FLOATS` would be good. And then rename the other one to `JSON_ROUND_FLOATS_MAX_DIGITS` or `JSON_ROUND_FLOATS_PRECISION`

Reply:
Should mocking be also conditional in tests?

Reply:
I don't think it's too important to have tests for the config values. They're generally simple enough that it doesn't matter